### DISCUSSION TITLE
NNN_TMPFILE including `cd` command is overly complicated?

### DISCUSSION THREAD

**Discussion by stelcodes** [jarun/nnn#1642] [Ideas]
I'm wondering why `NNN_TMPFILE` is written as a full `cd '/some-dir'` cd command when it seems like it would be better and simpler to just write `/some-dir` as the file contents. That way you wouldn't have to source that file from your shell, you could just do something like `cd "$(<$NNN_TMPFILE)"`. Is it because it's hard to get the shell quoting right if the directory name includes quotes? Big fan of nnn. Cheers!

**Comment by N-R-K** [comment]
I don't know why it was done this way. But it's kept this way so we don't break existing user's configs.

>Is it because it's hard to get the shell quoting right if the directory name includes quotes?

I don't think shell quoting would matter in the `cd "$(cat "$NNN_TMPFILE")"` case.
